src/scripts/main.js

```diff
@@ -2,10 +2,22 @@
 
 function waitFor(element, eventName) {
   // write your code here
+  return new Promise((resolve, reject) => {
```

Comment:
Consider adding error handling in the promise to handle cases where the event listener might fail to attach or if the element is null. This will make your code more robust.